rest/route/annotations.go

```diff
@@ -247,12 +247,27 @@ func (h *annotationByTaskPutHandler) Factory() gimlet.RouteHandler {
 func (h *annotationByTaskPutHandler) Parse(ctx context.Context, r *http.Request) error {
 	var err error
 	h.taskId = gimlet.GetVars(r)["task_id"]
+	taskExecutionsAsString := gimlet.GetVars(r)["task_execution"]
 	if h.taskId == "" {
 		return gimlet.ErrorResponse{
 			Message:    "task ID cannot be empty",
 			StatusCode: http.StatusBadRequest,
 		}
 	}
+	if taskExecutionsAsString == "" {
+		return gimlet.ErrorResponse{
+			Message:    "task execution cannot be empty",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	taskExec, err := strconv.Atoi(taskExecutionsAsString)
+	if err != nil {
+		return gimlet.ErrorResponse{
+			Message:    "cannot convert task execution to integer",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	h.annotation.TaskExecution = &taskExec
```

Comment:
TaskExecution doesn't need to be a pointer anymore. This was only necessary before because we needed to know if it wasn't provided, but now it will always be provided so we can just set it at 263 directly.

Reply:
Commented below. We still need it because it can be null in the body but provided in the url, or vice versa.